src/shogun2-core/shogun2-dao/src/main/java/de/terrestris/shogun2/dao/GenericHibernateDao.java

```diff
@@ -38,15 +38,45 @@ protected GenericHibernateDao(Class<E> clazz) {
 	@Autowired
 	private SessionFactory sessionFactory;
 
-	public Session getSession() {
+	/**
+	 * Obtains the current session.
+	 *
+	 * @return
+	 */
+	private Session getSession() {
 		return sessionFactory.getCurrentSession();
 	}
 
+	/**
+	 * Return the real object from the database. Returns null if the object does
+	 * not exist.
+	 *
+	 * @param id
+	 *
+	 * @see http://www.mkyong.com/hibernate/different-between-session-get-and-session-load/
+	 *
+	 * @return The object from the database or null if it does not exist
+	 */
 	public E findById(ID id) {
 		return (E) getSession().get(clazz, id);
 	}
 
 	/**
+	 * Return a proxy of the object (without hitting the database). This should
+	 * only be used if it is assumed that the object really exists and where
+	 * non-existence would be an actual error.
+	 *
+	 * @param id
+	 *
+	 * @see http://www.mkyong.com/hibernate/different-between-session-get-and-session-load/
+	 *
+	 * @return
+	 */
+	public E loadById(ID id) {
+		return (E) getSession().load(clazz, id);
```

Comment:
So will this method throw in case that the id isn't pointing to a valid existing object?

Or will subsequent actions fail loudly?